src/web/mjs/connectors/Qimiaomh.mjs

```diff
@@ -0,0 +1,84 @@
+import Connector from '../engine/Connector.mjs';
+
+export default class EpikManga extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'qimiaomh';
+        super.label = '奇妙漫画 (Qimiaomh)';
+        this.tags = [ '!test', 'manhua', 'chinese' ];
+        this.url = 'https://www.qimiaomh.com';
+
+        this.queryMangas = 'div.wrapper div.mt20 div.classification h2 a';
+        this.queryChapters = 'div.comic-content-list ul.comic-content-c li.tit a';
+        this.queryPages = 'div.cartoon div.loadimg img.man_img';
+    }
+
+    async _getMangasFromPage(page) {
+        const uri = new URL('/list-1------updatetime--' + page + '.html', this.url);
+        const request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        let checkData = await this.fetchDOM(request, 'div.paging ul li a');
+        let result = {
+            data: null,
+            run: true
+        };
+
+        result.data = data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.trim()
+            };
+        });
+
+        result.run = checkData.find(element => {
+            return element.href == "javascript:void(0)";
+        });
+
+        if(page === 1)result.run = null;
+
+        return result;
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            let result = await this._getMangasFromPage(page);
+            result.run ? run = false : mangaList.push(...result.data) ;
```

Comment:
As mentioned above, it is not intuitive to abort the loop when `result.run` is `true`

Reply:
Shouldn't the result still be pushed to the manga list for the current result?
If i read it correctly it just indicates that their is no next page, but the current page may still hold valid results...

Reply:
Maybe it is possible to just provide an empty manga list if there are no further results, this would make this whole tracking of `result.run` obsolete

Reply:
Yes, the current result is valid but the next result won't be valid.
example: 
open page 1000 by URL it will display the last page which is page 185 (no redirect)

Reply:
> Maybe it is possible to just provide an empty manga list if there are no further results, this would make this whole tracking of `result.run` obsolete, something like this:
> `result.length ? mangaList.push(...result) : run = false;`

will cause infinite loops

Reply:
Yes, this would require `_getMangasFromPage` to return an empty list, if the end is reached (instead of setting the run flag)

Reply:
yeah, either way still need a flag to determine the last page